src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -442,6 +442,49 @@ library GPv2Encoding {
         interaction.callData = interactionCallData;
     }
 
+    /// @dev Returns the number of order UIDs packed in a calldata byte array.
+    ///
+    /// This method reverts if the order UIDs data is malformed, i.e. the total
+    /// length is not a multiple of the length of a single order UID.
+    ///
+    /// @param orderUids The packed order UIDs.
+    /// @return count The total number of order UIDs packed in the specified
+    /// bytes.
+    function orderUidCount(bytes calldata orderUids)
+        internal
+        pure
+        returns (uint256 count)
+    {
+        require(
+            orderUids.length % ORDER_UID_LENGTH == 0,
+            "GPv2: malformed order UIDs"
+        );
+        count = orderUids.length / ORDER_UID_LENGTH;
+    }
+
+    /// @dev Returns a calldata slice to an order UID at the specified index.
+    ///
+    /// Note that this method does not check that the index is within the bounds
+    /// of the specified packed order UIDs, as reading calldata out of bounds
+    /// just produces 0's and will just decode to an invalid order UID that will
+    /// not correspond to an actual order.
```

Comment:
While true, I don't like very much this argument since it depends on the definition of the order uid and on the fact that the `validTo` is also hashed into the first part of the uid. I'd just say that the caller is supposed to have checked that the calldata is within the bounds.